### PR TITLE
W-11618110: return a new error handler in every request to the eh provider in shape validation

### PR DESCRIPTION
**Pull request by arielmirra**

### CODE REVIEW

File: amf-cli/shared/src/test/resources/validations/reports/model/draft-2019-unsupported-facets-warning.report [hunk 1]
```diff
@@ -1,18 +1,10 @@
 ModelId: file://amf-cli/shared/src/test/resources/validations/raml/draft-2019/draft-2019-unsupported-facets-warning.raml
 Profile: RAML 1.0
 Conforms: true
-Number of results: 7
+Number of results: 4
 
 Level: Warning
 
-- Constraint: http://a.ml/vocabularies/amf/validation#untranslatable-draft-2019-fields
```
Comment (tomsfernandez) [inline amf-cli/shared/src/test/resources/validations/reports/model/draft-2019-unsupported-facets-warning.report]: Why this ??

File: amf-cli/jvm/src/test/scala/amf/client/validation/JvmPayloadValidationTest.scala [hunk 2]
```diff
@@ -36,4 +35,23 @@ class JvmPayloadValidationTest extends ClientPayloadValidationTest with NativeOp
     positiveNumberReport.results.asSeq.head.message shouldBe "Can't divide by 0"
     zeroReport.results.asSeq.head.message shouldBe "Can't divide by 0"
   }
+
+  test("Validation order shouldn't affect conformance of each payload") {
+    val itemsShape = new NodeShape()
+    val shape      = new ArrayShape().withItems(itemsShape)
+    val validator  = payloadValidator(shape, `application/yaml`)
+
+    val validPayload   = "- {}"
```
Comment (tomsfernandez) [inline amf-cli/jvm/src/test/scala/amf/client/validation/JvmPayloadValidationTest.scala]: Was it only happening with syntax errors ?

File: amf-cli/shared/src/test/resources/validations/reports/model/draft-2019-unsupported-facets-warning.report (hunk 1 above)
Comment (arielmirra) [reply amf-cli/shared/src/test/resources/validations/reports/model/draft-2019-unsupported-facets-warning.report]: these validations seemed to be duplicated, saying the same thing twice.

File: amf-cli/jvm/src/test/scala/amf/client/validation/JvmPayloadValidationTest.scala (hunk 2 above)
Comment (arielmirra) [reply amf-cli/jvm/src/test/scala/amf/client/validation/JvmPayloadValidationTest.scala]: yes, it only seems to happen with syntax errors that reside in a multi-line payload

**Review by tomsfernandez — APPROVED** [review]

**Pull request merged**